### DISCUSSION TITLE
Can you provide this app as a Docker image so that it will be easy to self-host it?

### DISCUSSION THREAD

**Discussion by manojmandya07** [giscus/giscus#1179] [Q&A]
It would be difficult to troubleshoot while self hosting giscus app, so why can't you provide a docker image so that it will be easy to install and run the application.

**Comment by laymonage** [comment]
Contributions welcome!

**Reply by MrKragh** [reply]
I'm no expert here, more like Jon Snow, I know nothing! Anyways, even with my limited knowledge, you should be able to get something working like below.

My challenge is I know NOTHING about Node/Next.js, so in regards to production safety, I can't vouch for this.
One thing is certain, from my minimal testing I can only get it to work when I have the `.env` placed in root during Docker build, which **IS NOT SAFE** to have within Docker images. Tried with `--env-file` and mapping with `-v $(pwd)/.env:/app/.env` doesen't seem to do the trick. I suspect some environment variables are used during build. Again, I know nothing _nothing_.
Also, it somehow seems that signals isn't handled, so stopping the contain is clucky, use `docker stop giscus` to do so 🤷‍♂️ 

Maybe @laymonage can enrich, but seems like he has enough on his plate.

```Dockerfile
FROM node:18
WORKDIR /app

COPY package.json yarn.lock client.ts ./
RUN yarn install

COPY . .
RUN yarn build

CMD ["yarn", "start"]
```

```` shell
docker run --rm --name giscus -p 3000:3000 giscus
````

**Comment by shayanhoshyari** [comment]
it takes like 10 lines of code to build a docker image.
Ask chatgpt!

I empathize with the author not providing this. if you provide it, you have to support it. And I rather the author supports a small set of features great, rather than spreading himself thin.

on the author hand, what the users do when it comes to docker image, does not have to be supported by the author.
